### DISCUSSION TITLE
disable deletion or "safe delettion"

### DISCUSSION THREAD

**Discussion by massimiliano-dalcero** [outline/outline#4749] [General]
Hello :)

I'm new to outline, and I find it really interesting :)

I have a question: is it possible to disable the ability to delete "objects"?
because from what I understand, once something is deleted, it is no longer possible to recover it and I would like to prevent unpleasant accidents :)
Alternatively, is there something like "safe deletion"?

Thanks,
M.

**Comment by almereyda** [comment]
For compliance with European and national labour, tax and anti-fraud legislations[¹](https://en.wikipedia.org/wiki/Retention_period), many business artifacts cannot be deleted and must be conserved for a specific amount of years. In Germany this includes all electronic communication and documentation. It is called the [retention period](https://en.wikipedia.org/wiki/Retention_period), and a nice enterprise (hosted SaaS or on-premise) feature to have available.

Such a feature could be implemented through layered "recycling bins":

- a personal trash bin for users (unpublished documents)
- a trash bin for collections (published documents)

If those were to be cleared, and when retention is enabled, those could wander through additional layers of administrative retention collections:

- an administrative retention bin for collections
- an administrative retention bin for teams
- an administrative retention bin for the whole site

Else one would always have to refer to one's backups for restoration of data from the past.

**Comment by tommoor** [comment]
Hi @massimiliano-dalcero, all deletion is safe deletion – items are not deleted for 30 days unless deleted from the trash as well. This is the only retention period offered right now, it could be interesting to make that configurable and allow disabling full deletion entirely – as these are compliance features. they would not likely make it to the OSS edition though.
